app/src/main/java/mozilla/lockbox/view/ItemMutationFragment.kt

```diff
@@ -97,23 +104,17 @@ open class ItemMutationFragment : BackableFragment(), ItemMutationView {
         }
     }
 
-    open fun setupKeyboardFocus(view: View) {
-        view.inputUsername.onFocusChangeListener = View.OnFocusChangeListener { _, hasFocus ->
+    private fun focusChanges(view: EditText): Observable<Boolean> =
+        view.focusChanges().doOnNext { hasFocus ->
             if (!hasFocus) {
                 closeKeyboard()
             } else {
-                view.inputUsername?.setSelection(view.inputUsername?.length() ?: 0)
+                view.setSelection(view.length())
             }
         }
 
-        view.inputPassword.onFocusChangeListener = View.OnFocusChangeListener { _, hasFocus ->
-            togglePasswordVisibility.accept(Unit)
-            if (!hasFocus) {
-                closeKeyboard()
-            } else {
-                view.inputPassword?.setSelection(view.inputPassword?.length() ?: 0)
-            }
-        }
+    open fun setupKeyboardFocus(view: View) {
```

Comment:
Can this be removed now?

Reply:
The `Edit` and `Create` subclasses are still called. I don't understand the code enough to say if _those_ methods can be rendered obsolete.

I've added comment to this effect.